### PR TITLE
[one-cmds] Update one-import-bcq

### PR DESCRIPTION
**Pull request by llFreetimell**
Parent Issue : #4337
Full Draft : #4338

This commit will update `one-import-bcq` to use new python script.

ONE-DCO-1.0-Signed-off-by: Seok NamKoong <seok9311@naver.com>

### CODE REVIEW

File: compiler/one-cmds/one-import-bcq [hunk 1]
```diff
@@ -184,30 +184,41 @@ show_err_onexit()
 
 trap show_err_onexit ERR
 
-# generate temporary preserved pb file
-echo "${DRIVER_PATH}/preserve_bcq_info" --input_path ${INPUT_PATH} \
---output_path "${TMPDIR}/${MODEL_NAME}_preserved.pb"  > "${OUTPUT_PATH}.log"
+# Generate BCQ information metadata
+# If model has no BCQ information or invalid information, pb file is not changed.
+echo "${ROOT}/bin/generate_bcq_metadata" \
```
Comment (seanshpark) [inline compiler/one-cmds/one-import-bcq:189]: Where came the `ROOT` ?

Comment (llFreetimell) [reply compiler/one-cmds/one-import-bcq:189]: Ooops.... I thought that I removed all of them :(
That was wrongly copied from `tf2nnpkg`... thanks T.T

File: compiler/one-cmds/one-import-bcq [hunk 2]
```diff
@@ -184,30 +184,41 @@ show_err_onexit()
 
 trap show_err_onexit ERR
 
-# generate temporary preserved pb file
-echo "${DRIVER_PATH}/preserve_bcq_info" --input_path ${INPUT_PATH} \
---output_path "${TMPDIR}/${MODEL_NAME}_preserved.pb"  > "${OUTPUT_PATH}.log"
+# Generate BCQ information metadata
+# If model has no BCQ information or invalid information, pb file is not changed.
+echo "${DRIVER_PATH}/bin/generate_bcq_metadata" \
```
Comment (seanshpark) [inline compiler/one-cmds/one-import-bcq:189]: I think we should remove `bin/` :)

File: compiler/one-cmds/one-import-bcq [hunk 3]
```diff
@@ -184,30 +184,41 @@ show_err_onexit()
...
 
-# generate output_arrays automatically
+# Generate BCQ information nodes as output_arrays
+# If model has no BCQ information, output_arrays would be empty.
 echo "${DRIVER_PATH}/generate_bcq_output_arrays" \
---input_path "${TMPDIR}/${MODEL_NAME}_preserved.pb" \
---output_path "${TMPDIR}/${MODEL_NAME}_output_arrays.txt" > "${OUTPUT_PATH}.log"
+--input_path "${TMPDIR}/${MODEL_NAME}_withmeta.pb" \
+--metadata_path "${TMPDIR}/${MODEL_NAME}_metadata_arrays.txt" \
+--output_arrays_path "${TMPDIR}/${MODEL_NAME}_output_arrays.txt" > "${OUTPUT_PATH}.log"
```
Comment (seanshpark) [inline compiler/one-cmds/one-import-bcq:205]: Using `>` here will overwrite that was added in above. I didn't notice this for existing codes :)

File: compiler/one-cmds/one-import-bcq [hunk 4]
```diff
@@ -184,30 +184,41 @@ show_err_onexit()
...
+--output_path "${TMPDIR}/${MODEL_NAME}_withmeta.pb" \
+--output_arrays "${OUTPUT_ARRAYS}" > "${OUTPUT_PATH}.log"
 echo " " >> "${OUTPUT_PATH}.log"
 
-"${DRIVER_PATH}/preserve_bcq_info" --input_path ${INPUT_PATH} \
---output_path "${TMPDIR}/${MODEL_NAME}_preserved.pb" >> "${OUTPUT_PATH}.log" 2>&1
+"${DRIVER_PATH}/generate_bcq_metadata" \
+--input_path "${INPUT_PATH}" \
+--output_path "${TMPDIR}/${MODEL_NAME}_withmeta.pb" \
+--output_arrays "${OUTPUT_ARRAYS}"
```
Comment (seanshpark) [inline compiler/one-cmds/one-import-bcq:198]: There was `2>&1` in previous code but not removed. Is this intended?

File: compiler/one-cmds/one-import-bcq [hunk 5]
```diff
@@ -184,30 +184,41 @@ show_err_onexit()
 
 trap show_err_onexit ERR
 
-# generate temporary preserved pb file
-echo "${DRIVER_PATH}/preserve_bcq_info" --input_path ${INPUT_PATH} \
---output_path "${TMPDIR}/${MODEL_NAME}_preserved.pb"  > "${OUTPUT_PATH}.log"
+# Generate BCQ information metadata
+# If model has no BCQ information or invalid information, pb file is not changed.
+echo "${DRIVER_PATH}/generate_bcq_metadata" \
```
Comment (seanshpark) [inline compiler/one-cmds/one-import-bcq:189]: If you look at `one-import-tf`, `CONVERT_SCRIPT` is used to for the command and used for `echo` and execution.
Can you revise these lines like that?

Comment (llFreetimell) [reply compiler/one-cmds/one-import-bcq:189]: As another modification is needed, I will apply at this PR :)

File: compiler/one-cmds/one-import-bcq [hunk 6]
```diff
@@ -184,30 +184,37 @@ show_err_onexit()
...
+$METAGEN_SCRIPT >> "${OUTPUT_PATH}.log" 2>&1
+
+# Generate BCQ information nodes as output_arrays
+# If model has no BCQ information, output_arrays would be empty.
+OUTARR_GEN_SCRIPT="${DRIVER_PATH}/generate_bcq_output_arrays "
+OUTARR_GEN_SCRIPT+="--input_path ${TMPDIR}/${MODEL_NAME}_withmeta.pb "
+OUTARR_GEN_SCRIPT+="--metadata_path ${TMPDIR}/${MODEL_NAME}_metadata_arrays.txt "
+OUTARR_GEN_SCRIPT+="--output_arrays_path ${TMPDIR}/${MODEL_NAME}_output_arrays.txt "
+
+echo ${OUTARR_GEN_SCRIPT} > "${OUTPUT_PATH}.log"
```
Comment (seanshpark) [inline compiler/one-cmds/one-import-bcq:205]: I think this should use `>>` to append :)

**Review by seanshpark — APPROVED** [review]
LGTM thank you

**Pull request merged**